src/compiler/parser.ts

```diff
@@ -3337,7 +3337,10 @@ namespace ts {
         function parseAwaitExpression() {
             const node = <AwaitExpression>createNode(SyntaxKind.AwaitExpression);
             nextToken();
-            node.expression = parseSimpleUnaryExpression();
+            node.expression = token === SyntaxKind.AwaitKeyword
+                // Forbid `await await`

```

Comment:
In the [spec](https://tc39.github.io/ecmascript-asyncawait/#prod-UnaryExpression) it looked like 'await' was specifically exempted from following. Maybe I read that wrong.